multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -18,38 +18,92 @@ package multicluster
 
 import (
 	"context"
+	"fmt"
+	"sync"
 
+	"github.com/go-logr/logr"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/internal"
 )
 
 // ClusterSetReconciler reconciles a ClusterSet object
+// There will be one controller running in each namespace of the leader for multiple cluster-set support
+// so each controller will only be handling a single cluster set in the given namespace.
 type ClusterSetReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	mutex    sync.Mutex
+	Scheme   *runtime.Scheme
+	Log      logr.Logger
+	IsLeader bool
+
+	clusterSetConfig *multiclusterv1alpha1.ClusterSet
+	clusterSetID     common.ClusterSetID
+	clusterID        common.ClusterID
+
+	// These fields are only applicable on leader cluster
+	InboundLeaderMgr internal.LocalClusterManager
```

Comment:
Why not just name it "LeaderMgr"?

Reply:
I changed it to LocalClusterManager now, I missed changing it after renaming the class, hope this is ok?